scripts/setup_ubuntu.sh

```diff
@@ -2,6 +2,19 @@
 
 set -xeuo pipefail
 
+#
+# Install bootstrap requirements
+#
+
+apt update --quiet
+apt install -yq lsb-release curl software-properties-common
+
+NO_UPDATE=""
+if apt-add-repository  --help | grep -q -e "--no-update"; then
+  NO_UPDATE="--no-update"
+fi
```

Comment:
I'm guessing this is the 16.04 hack you mentioned. Possibly ignorant question, would this be better gated with `$RELEASE = 'bionic'` or something similar?

Reply:
This actually isn't the hack. the hack is the ability to build python without it's extension modules and to load the `katana` module without those extensions (Handling `RequirementError` and `ModuleNotFoundError` resp).

I did a grep here because I was lazy more than anything, and apt didn't list when `--no-update` was added, so I didn't know which version to check for. So I just check for the feature. ;-)